presto-product-tests/src/main/java/io/prestosql/tests/hive/TestHiveTransactionalTable.java

```diff
@@ -166,4 +182,66 @@ private static String hiveTableProperties(TransactionalTableType transactionalTa
                 bucketingType.getHiveTableProperties().stream())
                 .collect(joining(",", "TBLPROPERTIES (", ")"));
     }
+
+    private static void compactTableAndWait(CompactionMode compactMode, String tableName, Duration timeout)
+    {
+        onHive().executeQuery(format("ALTER TABLE %s COMPACT '%s'", tableName, compactMode.name()));
+
+        Set<String> tableCompactionsIds = getTableCompactions(compactMode, tableName)
+                .map(row -> row.get("compactionid"))
+                .collect(toUnmodifiableSet());
+
+        assertEventually(timeout, () -> {
+            Set<String> completedCompactionsIds = getTableCompactions(compactMode, tableName)
+                    .filter(row -> row.get("state").equals("succeeded"))
```

Comment:
Is it possible that `tableCompactionIds` obtained above will contain a compaction which is in final, but no`succeeded` state (e.g. `failed`)? Then we will never satisfy the assertion.

Not sure if that is something to worry about. Possibly you may additionally list compactionIds before `ALTER TABLE` is called. And then ignore those later in assertion. Would that work?

Reply:
Done.